predict/predict_baseline_models.R

```diff
@@ -0,0 +1,208 @@
+# Train and test baseline models to predict MB subgroups
+# Steven Foltz, updated 2025-02
+
+# set some variables that guide what downstream steps are followed
+create_models <- TRUE # train new models (if FALSE, reads existing models from file)
+overwrite <- TRUE # if create_models is also TRUE, overwrite existing models file
+seed <- 44 # set initial seed for run_many_models() and before stochastic plots
+n_repeats <- 10 # set number of times to repeat in run_many_models()
+n_cores <- 3 # set number of cores to use in run_many_models()
+ah_date <- "2022-10-30"
+
+# Rationale ----
+
+#We want to use single sample prediction models to predict tumor subgroup in medulloblastoma (MB) patient samples.
+#Our gene expression data comes from multiple studies, and different studies were generated using different platforms (array and RNA-seq).
+
+#Our goals are to train (kTSP, RF, LASSO) or use (MM2S, medullopackage) different prediction models and to assess model performance on a per-platform and per-subgroup basis.
+#Ideally, models would perform well regardless of the platform or subgroup of the sample, but differences in sample size mean some groups may be underrepresented in the data available for training.
+#Inverse-weighting may boost performance for smaller groups, and we can test this in kTSP and RF models using parameters for weighted (w) and unweighted (unw) analyses.
+
+#To train and test our models, we use the `run_many_models()` function.
+#This function allows us to specify what model types we want to analyze, various model parameters, and segregate samples into training/testing sets via a random seed.
+#We train and test each model a number of times (e.g. 10 repeats) using different training/testing sets to understand the range of model performance as measured by Kappa and Balanced Accuracy.
+#Training/testing sets segregate at the study level, rather than the sample level.
+
+# Setup ----
+
+# Source code and libraries
+source(here::here("utils/color_schemes.R"))
+source(here::here("utils/convert_gene_names.R"))
+source(here::here("utils/modeling.R"))
+
+# Define directories and input/output file paths
+processed_data_dir <- here::here("processed_data")
+models_dir <- here::here("models")
+
+bulk_genex_filepath <- file.path(processed_data_dir, "bulk_genex.tsv")
+bulk_metadata_filepath <- file.path(processed_data_dir, "bulk_metadata.tsv")
+
+baseline_filepath <- file.path(models_dir, "baseline.rds")
+
+# check that existing baseline model file will not be overwritten if overwrite is FALSE
+if (create_models &
+    (file.exists(baseline_filepath)) &
+    !overwrite) {
+
+  stop("Model output file already exists and overwrite is set to FALSE in analysis_notebooks/baseline.Rmd.")
+
+}
+
+# check that files exist if create_models is FALSE
+if (!create_models & !file.exists(baseline_filepath)) {
+
+  stop("Model output file does not exist and create_models is set to FALSE in analysis_notebooks/baseline.Rmd.")
```

Comment:
I assume having the script name will be handy when one runs all the bulk experiments.

```suggestion
  stop("Model output file does not exist and create_models is set to FALSE in `predict/predict_baseline_models.R.")
```

Reply:
Updated here and one other file check 👍